### DISCUSSION TITLE
Remove specified role via command

### DISCUSSION THREAD

**Discussion by freshefisch** [Rapptz/discord.py#6513] [Q&A]
I would like to have a command that you can remove the role "testrole" with.

It should look something like this.

```
@bot.command
async def remove_testrole(ctx):
   await ctx.remove_roles(testrole)
```

Can anyone help me how I should do this?
Thanks in advance!

**Comment by Nemika-Haj** [comment]
You can add a parameter and parse it as a role, like so;
```py
@bot.command
async def remove_testrole(ctx, testrole:discord.Role)
```
This will accept both role ID's, names, and mentions.

**Reply by freshefisch** [reply]
> 
> 
> You can add a parameter and parse it as a role, like so;
> 
> ```python
> @bot.command
> async def remove_testrole(ctx, testrole:discord.Role)
> ```
> 
> This will accept both role ID's, names, and mentions.

I get an error that 'ctx has no attribute remove_roles'. So how do I then remove the role from the user that sent the message?

![python error](https://user-images.githubusercontent.com/62943210/110217520-b329b480-7eb4-11eb-9a28-3efefd9b3c18.png)

The code:

```
@bot.command()
async def remove_role(ctx, testrole:discord.Role):
  await ctx.remove_roles(testrole)
```
What I entered:

_$remove_role testrole_

**Reply by Nemika-Haj** [reply]
Obviously, you should change `ctx.remove_roles` to `ctx.author.remove_roles`

`ctx` alone returns `Context` and `ctx.author` returns the command author. For such easy-to-fix problems, please consult the documentation; https://discordpy.readthedocs.io/en/latest/
